### DISCUSSION TITLE
class-validator constructs in applyInputTypesEnhanceMap not working with mutations

### DISCUSSION THREAD

**Discussion by Morriz** [MichalLytek/typegraphql-prisma#319] [Q&A]
**Describe the Bug**
Due to prisma wanting mutations to be set with `someProp: { set: $someProp }`, the class-validator methods don't receive the correct value. Is it possible to detect this shape and inject the value of `$someProp` instead?

**To Reproduce**

```
export const QUERY_GET_USER = gql`
  query User($id: String!) {
    user(where: { id: $id }) {
      id
      name
      email
      company
    }
  }
`

export const MUTATION_UPDATE_USER = gql`
  mutation UpdateUser(
    $id: String!
    $name: String
    $company: String
  ) {
    updateUser(
      where: { id: $id }
      data: {
        name: { set: $name }
        company: { set: $company }
      }
    ) {
      updatedAt
    }
  }

...

applyArgsTypesEnhanceMap({
  UpdateUserArgs: {
    fields: {
      data: [ValidateNested()],
    },
  },
})

applyInputTypesEnhanceMap({
  UserUpdateInput: {
    fields: {
      _all: [Allow()],
      company: [MinLength(10)],
    },
  },
})
```

**Expected Behavior**

The value to be validated and no error thrown.

**Logs**
If applicable, add some console logs to help explain your problem.
You can paste the errors with stack trace that were printed when the error occurred.

**Environment (please complete the following information):**

- OS: MacOS
- Node: v16.14.2
- `typegraphql-prisma` version: 0.20.4
- Prisma version: 3.15.2
- TypeScript version: 4.6.4

**Additional Context**

**Comment by MichalLytek** [comment]
You need to wait for #290 #32

**Reply by Morriz** [reply]
Aha. But wait for what? I see nothing addressed in either of those issues.

**Reply by Morriz** [reply]
Is it not possible to detect that shape and inject the correct value instead?

**Reply by MichalLytek** [reply]
The correct value is `set` as this is what Prisma requires.
Removing `set` and having simple input is #32. Please read more carefully.

**Reply by Morriz** [reply]
Ok, I see it is part of type-graphql and that has no awareness of prisma...

**Reply by Morriz** [reply]
> When their API becomes stable and this generator package will have other features done, we can investigate the possibility to make some changes in the inputs shape, like omit set for string fields.

I think you are referring to that then
